### ISSUE TITLE
Incorrect output when playing back nano sessions

### ISSUE THREAD

**Issue by robmoss** [asciinema/asciinema-player#207]
I've been using asciinema to record some git examples. I use nano as my commit message editor, and noticed that while `asciinema play` plays back the sessions as expected, asciinema-player (version 3.0.1) doesn't correctly play back the commit messages I record in nano.

Below is a minimal example where I open a two-line text file, cut the second line, and paste it above the first line. When played back with asciinema-player, the editor title bar disappears when I paste the second line. In longer recordings, when I'm editing text in nano, asciinema-player sometimes displays text that was printed to the terminal before nano was opened. So I guess there's an escape code that's not being handled correctly, perhaps?

Please let me know if there are any other useful details I can provide.

```text
{"version": 2, "width": 80, "height": 24, "timestamp": 1673489926, "env": {"SHELL": "/usr/bin/bash", "TERM": "xterm-256color"}}
[0.033791, "o", "\u001b]2;(x1::~)\u0007\u001b[32;1m(x1::~)\u001b[0m "]
[0.585084, "o", "n"]
[0.729152, "o", "a"]
[0.809063, "o", "n"]
[0.865118, "o", "o"]
[0.961145, "o", " "]
[1.073137, "o", "n"]
[1.161112, "o", "a"]
[1.241035, "o", "n"]
[1.289063, "o", "o"]
[1.481012, "o", "-"]
[1.608512, "o", "e"]
[1.768764, "o", "x"]
[1.849084, "o", "a"]
[1.905123, "o", "m"]
[2.033063, "o", "p"]
[2.112923, "o", "l"]
[2.209151, "o", "e"]
[2.313148, "o", "."]
[2.400998, "o", "t"]
[2.62507, "o", "x"]
[2.713115, "o", "t"]
[2.896852, "o", "\r\n"]
[2.900342, "o", "\u001b[?2004h\u001b[?1049h\u001b[22;0;0t\u001b[1;24r\u001b(B\u001b[m\u001b[4l\u001b[?7h\u001b[39;49m\u001b[?1h\u001b=\u001b[?1h\u001b=\u001b[?25l"]
[2.900862, "o", "\u001b[39;49m\u001b(B\u001b[m\u001b[H\u001b[2J\u001b[22;34H\u001b(B\u001b[0;7m[ Reading... ]\u001b(B\u001b[m"]
[2.900907, "o", "\u001b[22;33H\u001b(B\u001b[0;7m[ Read 2 lines ]\u001b(B\u001b[m"]
[2.901376, "o", "\u001b[H\u001b(B\u001b[0;7m  GNU nano 4.8                    nano-example.txt                              \u001b[1;79H\u001b(B\u001b[m"]
[2.901413, "o", "\r\u001b[23d\u001b(B\u001b[0;7m^G\u001b(B\u001b[m Get Help  \u001b(B\u001b[0;7m^O\u001b(B\u001b[m Write Out \u001b(B\u001b[0;7m^W\u001b(B\u001b[m Where Is  \u001b(B\u001b[0;7m^K\u001b(B\u001b[m Cut Text  \u001b(B\u001b[0;7m^J\u001b(B\u001b[m Justify   \u001b(B\u001b[0;7m^C\u001b(B\u001b[m Cur Pos\r\u001b[24d\u001b(B\u001b[0;7m^X\u001b(B\u001b[m Exit\u001b[14G\u001b(B\u001b[0;7m^R\u001b(B\u001b[m Read File \u001b(B\u001b[0;7m^\\\u001b(B\u001b[m Replace   \u001b(B\u001b[0;7m^U\u001b(B\u001b[m Paste Text\u001b(B\u001b[0;7m^T\u001b(B\u001b[m To Spell  \u001b(B\u001b[0;7m^_\u001b(B\u001b[m Go To Line\r\u001b[22d"]
[2.901439, "o", "\u001b[2d"]
[2.90146, "o", "\u001b[39;49m\u001b(B\u001b[mFirst line\r\u001b[3dSecond line\r\u001b[A\u001b[?12l\u001b[?25h"]
[3.616855, "o", "\u001b[?25l\u001b[3d\u001b[?12l\u001b[?25h"]
[4.169075, "o", "\u001b[?25l"]
[4.169373, "o", "\u001b[1;71H\u001b(B\u001b[0;7mModified\u001b(B\u001b[m"]
[4.169926, "o", "\r\u001b[3d\u001b[K\u001b[22d\u001b[K\u001b[3d\u001b[?12l\u001b[?25h"]
[4.593045, "o", "\u001b[?25l"]
[4.593334, "o", "\u001b[A\u001b[?12l\u001b[?25h"]
[5.233079, "o", "\u001b[?25l"]
[5.233146, "o", "\u001b7\u001b[2;22r\u001b8\u001bM\u001b[1;24r\u001b[2;1HSecond line\r\u001b[3d\u001b[?12l\u001b[?25h"]
[5.593147, "o", "\u001b[?25l"]
[5.593495, "o", "\u001b[23;14H       \u001b(B\u001b[0;7mM-D\u001b(B\u001b[m DOS Format      \u001b(B\u001b[0;7mM-A\u001b(B\u001b[m Append          \u001b(B\u001b[0;7mM-B\u001b(B\u001b[m Backup File\u001b[24;2H\u001b(B\u001b[0;7mC\u001b(B\u001b[m Cancel           \u001b(B\u001b[0;7mM-M\u001b(B\u001b[m Mac Format      \u001b(B\u001b[0;7mM-P\u001b(B\u001b[m Prepend         \u001b(B\u001b[0;7m^T\u001b(B\u001b[m To Files\u001b[K\r\u001b[22d\u001b(B\u001b[0;7mFile Name to Write: nano-example.txt                                            \u001b[22;1H\u001b(B\u001b[m\u001b[22;37H\u001b[?12l\u001b[?25h"]
[5.929158, "o", "\u001b[?25l"]
[5.929496, "o", "\b\b\b\b\u001b[1K \u001b(B\u001b[0;7m[ Writing... ]\u001b(B\u001b[m\u001b[K"]
[5.929654, "o", "\u001b[1;71H\u001b(B\u001b[0;7m        \u001b(B\u001b[m\u001b[22;32H\u001b(B\u001b[0;7m[ Wrote 2 lines ]\u001b(B\u001b[m"]
[5.929833, "o", "\u001b[23;14H\u001b(B\u001b[0;7m^O\u001b(B\u001b[m Write Out \u001b(B\u001b[0;7m^W\u001b(B\u001b[m Where Is  \u001b(B\u001b[0;7m^K\u001b(B\u001b[m Cut Text  \u001b(B\u001b[0;7m^J\u001b(B\u001b[m Justify   \u001b(B\u001b[0;7m^C\u001b(B\u001b[m Cur Pos\u001b[24;2H\u001b(B\u001b[0;7mX\u001b(B\u001b[m Exit      \u001b(B\u001b[0;7m^R\u001b(B\u001b[m Read File \u001b(B\u001b[0;7m^\\\u001b(B\u001b[m Replace   \u001b(B\u001b[0;7m^U\u001b(B\u001b[m Paste Text\u001b(B\u001b[0;7m^T\u001b(B\u001b[m To Spell  \u001b(B\u001b[0;7m^_\u001b(B\u001b[m Go To Line\r\u001b[22d\u001b[3d\u001b[?12l\u001b[?25h"]
[6.888845, "o", "\u001b[?25l"]
[6.889121, "o", "\u001b[22d\u001b[J\u001b[24d\u001b[?12l\u001b[?25h\u001b[24;1H\u001b[?1049l\u001b[23;0;0t\r\u001b[?1l\u001b>\u001b[?2004l"]
[6.889743, "o", "\u001b]2;(x1::~)\u0007\u001b[32;1m(x1::~)\u001b[0m "]
[7.288939, "o", "exit\r\n"]
```

**Comment by sickill** [comment]
Indeed it looks like a bug in terminal emulation. Thanks for reporting and for the minimal example :ok_hand:

**Comment by robmoss** [comment]
No worries! I was thoroughly confused at first and I tried a few different things before deciding it was probably an issue in asciinema-player. It's an awesome tool, I've been having fun figuring out how to make best use of it in some of our training materials :smile:

**Comment by sickill** [comment]
Thanks for your kind words, I'm glad you like the project :)

I found the issue and fixed it here: https://github.com/asciinema/vt-rs/commit/5f8ee15ba34e0a2c3c7466bbe3f639b919a0e3aa

I'll be releasing new bugfix version of the player with updated VT module soon.

**Comment by robmoss** [comment]
Brilliant, thank you very much!

**Comment by sickill** [comment]
Just released 3.1.0 with the fix.

**Comment by robmoss** [comment]
Brilliant, thank you very much!
